cmake/pcl_targets.cmake

```diff
@@ -169,8 +169,8 @@ macro(PCL_SUBSUBSYS_DEPEND _var _parent _name)
     endif()
     if(SUBSUBSYS_EXT_DEPS)
       foreach(_dep ${SUBSUBSYS_EXT_DEPS})
-        string(TOUPPER "${_dep}_found" EXT_DEP_FOUND)
```

Comment:
Are you sure this won't lead to problems? E.g. for `OpenGL` or `openni`, where the variables are `OPENGL_FOUND` and `OPENNI_FOUND` (all uppercase). You changed it for `Qt5` (`Qt5_FOUND`), right?

Reply:
Ah, this might be true. So we should check for either xx_FOUND or XX_FOUND - maybe?

Reply:
Either that or we make sure that there is always an all-uppercase variable available (e.g. check if `Qt5_FOUND` is true, then define `QT5_FOUND` in the top-most CMakeLists.txt). Not sure which alternative is less error-prone?

Reply:
Here its only one place. Else you'll have to do it multiple places?

There names also has to be right when adding EXT_DEPS ie. VTK vs vtk.

Reply:
> There names also has to be right when adding EXT_DEPS ie. VTK vs vtk.

Why is that?

However we decide, we should apply the same for `PCL_SUBSYS_DEPEND` since the `string(TOUPPER ...` appears there, too.

Reply:
Right now when searched for Qt5, it will define `Qt5_FOUND`.
But our PCL_(SUB)SUBSYS_DEPEND will capitalize it to `QT5_FOUND` which would be false.
With my latests addition, both Qt5_FOUND or QT5_FOUND is checked.
However, if I add EXT_DEPTS QT5 - it will fail as only `Qt5_FOUND `is defined.

So maybe it's better to add capitalizing of XX_FOUND after each find_package. Though it has to be multiple places.

Reply:
I generally thought that find_package_handle_standard_args(xx) would return xx_FOUND, but with Openni its explicit set that OPENNI_FOUND is the variable to use. So there are many exceptions to this rule 😢 

https://cmake.org/cmake/help/latest/module/FindPackageHandleStandardArgs.html

Seems that FOUND_VAR is obsolute and both `xx_FOUND `and `XX_FOUND `should be set now.

Reply:
I have added a QT5_FOUND variable. Seems it was the only one making trouble.